conda_forge_tick/migrators/core.py

```diff
@@ -443,6 +445,51 @@ def __init__(
         self.top_level = top_level or set()
         self.cycles = set(chain.from_iterable(cycles or []))
 
+    def all_predecessors_issued_and_stale(self, attrs: "AttrsTypedDict") -> bool:
+        # Check if all upstreams have been issue and are stale
+        for node in self.graph.predecessors(attrs["feedstock_name"]):
+            try:
+                payload = self.graph.nodes[node]["payload"]
+            except KeyError as e:
+                print(node)
+                raise e
+
+            # we don't need to look at archived feedstocks
+            # they are always "migrated"
+            if payload.get("archived", False):
+                continue
```

Comment:
Is it worthwhile pulling this out as a generator that we walk through, to reduce code copy paste?

Reply:
yup we can do that!